app/services/create_at_desk.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CreateAtDesk
+  def initialize(screening_id, seats)
+    @screening_id = screening_id
+    @seats = seats
+    @reservation = Reservation.new(screening_id:, status: :confirmed)
+  end
+
+  def call
+    return false unless seats_selected?
+
+    ActiveRecord::Base.transaction do
+      @reservation.save!
+      create_tickets
+    end
+  end
+
+  private
+
+  attr_reader :seats
+
+  def create_tickets
+    seats.each do |seat|
+      @reservation.tickets.create(seat:)
+    end
+  end
+
+  def seats_selected?
+    !seats.nil?
```

Comment:
Where are you filtering seats? I see that is [a array of ids](https://github.com/danyflorczak/Monte_Cinema/pull/13/files#diff-578962dfd4d839cb1a1bcc2d97d270868f9fdbe3b5343d70084fa598b470dd36R31) so I assume that this id are filtered somewhere.

Reply:
I'm filtering them inside of screening model but I'll be changing that to some service/validators to separate that from models